tap-version-14-specification.md

```diff
@@ -0,0 +1,741 @@
+whitespace, it SHOULD be discarded by the Harness.  For example, these
+test lines should all be identical:
+
+```
+ok 5 some description
+ok 5 - some description
+ok 5 - some description # and some comment
+```
+
+The harness may do whatever it wants with the description.
+
+- Directive
+
+The test point may include a directive, following a hash on the test
+line.
+
+There are currently two directives allowed: TODO and SKIP. These are
+discussed below.  Any other directive is ignored.
+
+#### Test Line Summary
+
+To summarize:
+
+- ok/not ok (required)
+- Test number (recommended)
+- Description (recommended)
+- Directive (only when necessary)
+
+The regular expression pattern for a test line is:
+
+```
+/^(not )?ok( +[1-9][0-9]*)?( +(- +)[\u0001-\u0009\u000B-\u000C\u000E-\u0022\u0024-\u00FF]+)?( #(.*))?$/
+```
+
+### YAML blocks
+
+If a test line is immediately followed by a 2-space indented block
+beginning with the line `/^ {2}---$/` and ending with the line
+`/^ {2}\.\.\.$/` that block will be interpreted as an inline YAML
+document.
+
+The YAML encodes a data structure that provides more detailed
+information about the preceding failed test.  The YAML document is
+indented with 2 spaces to make it visually distinct from the
+surrounding test results and to make it easier for the parser to
+recover if the trailing `/^ {2}\.\.\.$/` terminator is missing.
+
+For example:
+
+```
+not ok 3 Resolve address
+  ---
+  message: "Failed with error 'hostname peebles.example.com not found'"
+  severity: fail
+  data:
+    got:
+      hostname: 'peebles.example.com'
+      address: ~
+    expected:
+      hostname: 'peebles.example.com'
+      address: '85.193.201.85'
+  ...
+```
+
+The corresponding data structure in JSON would look like this:
+
+```json
+{
+  "message": "Failed with error 'hostname peebles.example.com not found'",
+  "severity": "fail",
+  "data": {
+    "got": {
+      "hostname": "peebles.example.com",
+      "address": null
+    },
+    "expected": {
+      "hostname": "peebles.example.com",
+      "address": "85.193.201.85"
+    }
+  }
+}
+```
+
+If a line occurs within the YAML document that is not indented, or if
+the TAP stream ends before the `/^ {2}\.\.\.$/` terminator line, or if the
+YAML document does not parse as YAML, then it MUST be treated as
+non-compliant data, and handled by the Harness the same as any other
+non-TAP lines.
+
+Currently (2015-10-06) the format of the data structure represented by
+a YAML block has not been standardized.  It is likely that a future
+version of TAP will standardize meanings for some of the fields in the
+YAML diagnostics.
+
+In order to adequately interpret and report the data within a test
+output's YAML block, a full YAML parser is required.  Specifying the
+YAML language is outside the scope of this document.  Harneses MAY
+ignore YAML blocks entirely if the inclusion of a full YAML parser
+would be overly difficult.
+
+## DIRECTIVES
```

Comment:
The casing here seems a little odd (I know that is how the current version is however I think I would change it for everything being sentence case or upper first)